### PR TITLE
[5.1] HTML Mail template images

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue ##40678 .

### Summary of Changes
When sending html mails created with com_mails it is possible to include images.

The function convertRelativeToAbsoluteUrls ensures that the images have absolute references so that they will be loaded in the email. However if the image has been added to the email using the media manager it will include loading="lazy" and data-path=

This pr strips any references to loading and data-path

https://www.caniemail.com/search/?s=loading

### Testing Instructions
Set the mail templates to use html
Add an image to one of the mail templates - i suggest the Global Configuration: Test Mail
Trigger the mail and check it

### Actual result BEFORE applying this Pull Request
The code for the image in the email will be
`
`

### Expected result AFTER applying this Pull Request
The code for the image in the email will be
`
`

The originakl issue report was classified as a bug therefore it should be 5.0 but it seems more like a feature so I made it against 5.1 but happy to change that.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/Mail/MailHelper.php [hunk 1]
```diff
@@ -216,20 +216,28 @@ public static function convertRelativeToAbsoluteUrls($content)
...
+                    // Define a regular expression pattern for matching relative URLs in the specified attribute
+                    $regex = '#\s' . $attribute . '"(?!/|' . $protocols . '|\#|\')([^"]*)"#m';
+
+                    // Replace relative URLs with absolute URLs using the siteUrl variable
+                    $content = preg_replace($regex, ' ' . $attribute . '"' . $siteUrl . '$1"', $content);
+                }
                 self::checkContent($content);
             }
         }
-
```
Comment (Quy) [inline libraries/src/Mail/MailHelper.php]: Please revert.

Comment (brianteeman) [reply libraries/src/Mail/MailHelper.php]: Shoudnt drone phpcs report if there is an issue with this?

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 9e627d1e2fe059bb424ee3cd124777fee78bd453<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42448">issues.joomla.org/tracker/joomla-cms/42448</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42448">issues.joomla.org/tracker/joomla-cms/42448</a>.</sub>

File: libraries/src/Mail/MailHelper.php (hunk 1 above)
Comment (LadySolveig) [reply libraries/src/Mail/MailHelper.php]: @Quy you set to RTC but this is not resolved yet?

Comment (LadySolveig) [reply libraries/src/Mail/MailHelper.php]: @brianteeman could you please revert. I would really like to merge this.

**Comment by brianteeman** [comment]
thx

**Pull request merged**